dask/dataframe/tests/test_groupby.py

```diff
@@ -3190,3 +3191,88 @@ def test_groupby_slice_getitem(by, slice_key):
     # column projection after read_parquet etc
     assert hlg_layer(got.dask, "getitem")
     assert_eq(expect, got)
+
+
+@pytest.mark.parametrize(
+    "func",
+    [
+        "min",
+        "max",
+        "sum",
+        "prod",
+        "first",
+        "last",
+        "idxmin",
+        "idxmax",
+    ],
+)
+@pytest.mark.parametrize(
+    "numeric_only",
+    [None, True, False],
+)
+def test_groupby_numeric_only_supported(func, numeric_only):
+    pdf = pd.DataFrame({"A": [1, 1, 2], "B": [3, 4, 3], "C": ["a", "b", "c"]})
+
+    ddf = dd.from_pandas(pdf, npartitions=3)
+
+    kwargs = {} if numeric_only is None else {"numeric_only": numeric_only}
+
+    expect_type_error = False
+    expect_future_warning = False
+    expected = None
+    try:
+        with warnings.catch_warnings(record=True) as w:
+            expected = getattr(pdf.groupby("A"), func)(**kwargs)
+            if len(w) > 0 and w[0].category == FutureWarning:
```

Comment:
Might also be good to do a simple check that this is the warning message we want. For example, check that "numeric_only" is in the warning message